### DISCUSSION TITLE
Symbols vs Sigils

### DISCUSSION THREAD

**Discussion by ghost** [h3rald/min#94] [Q&A] [answered]
I didn't understand what difference of symbols and sigils have, from doc.
Could you explain the difference with some examples?

**Comment by h3rald** [comment, marked as answer]
Right so... A **symbol** is basically a word that when pushed on the stack it is evaluated and manipulated the stack -- this is fairly easy to understand and let's say it is similar to a function in non-concatenative programming languages.

A **sigil** is a prefix you put in front of a string, and it is just a shorthand for calling a symbol with one string input value.

System sigils typically use special characters and if you think about it you use them all the time:

     1 :test

Instead of:

     1 "test" define


Or:


     *my-module/my-symbol

Instead of:

     "my-module/my-symbol" invoke

When the string input value contains spaces, you have to use double quote as you would with a string, so:

     !"ls -al"

Then there's user-defined sigils. User-defined sigils typically are longer than one character and they can only contain valid symbol characters (so letters, numbers, underscores, dashes, etc) so typically you'd want to use them with double-quoted strings really.

You could for example define a sigil for, I don't know, whatever takes a string input value:

     (strip) 'strip :define-sigil

      strip" aaaaa " ; "aaaaa"

Basically, sigils are nothing more than syntactic sugar 😊

**Reply by ghost** [reply]
Yes, thanks. 😄
